dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerTest.kt

```diff
@@ -402,4 +420,182 @@ class CompanyDataControllerTest {
+            )
+        }
+        assertTrue(exception.statusCode == 403, "The exception should indicate unauthorized access (HTTP 403)")
+    }
+*/
+    @Test
+    fun `check that the dataUploader can patch contactDetails and website if the company does not have companyOwner`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val companyId = uploadInfo.actualStoredCompany.companyId
+        val patchObject = CompanyInformationPatch(
+            companyContactDetails = setOf("New-Email-1@example.com", "New-Email-2@example.com"),
+            website = "New-Website",
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+        val updatedCompany = apiAccessor.companyDataControllerApi.patchCompanyById(
+            companyId,
+            patchObject,
+        )
+        assertEquals(
+            patchObject.companyContactDetails!!, updatedCompany.companyInformation.companyContactDetails,
+            "The company contact details should have been updated",
+        )
+        assertEquals(
+            patchObject.website!!, updatedCompany.companyInformation.website,
+            "The company website should have been updated",
+        )
+    }
+
+    @Test
+    fun `check that the dataUploader cannot patch contactDetails or website if the company has a companyOwner`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val ownerId = UUID.fromString("18b67ecc-1176-4506-8414-1e81661017ca")
+        val originalCompany = uploadInfo.actualStoredCompany
+        val companyId = originalCompany.companyId
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyRolesControllerApi.assignCompanyRole(
+            CompanyRole.CompanyOwner,
+            UUID.fromString(companyId),
+            ownerId,
+        )
+
+        val patchObject = CompanyInformationPatch(
+            companyContactDetails = setOf("New-Email-3@example.com"),
+            website = "New-Website-2",
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+
+        val exception = assertThrows<ClientException> {
+            apiAccessor.companyDataControllerApi.patchCompanyById(
+                companyId,
+                patchObject,
+            )
+        }
+
+        assertTrue(exception.statusCode == 403, "The exception should indicate unauthorized access (HTTP 403)")
+
+        val companyAfterAttempt = apiAccessor.companyDataControllerApi.getCompanyById(companyId)
+        assertEquals(
+            originalCompany.companyInformation.companyContactDetails,
+            companyAfterAttempt.companyInformation.companyContactDetails,
+            "The company contact details should not have been updated",
+        )
+    }
+
+    @Test
+    fun `check that the dataUploader cannot patch unallowed fields`() {
```

Comment:
schreib lieber "check that Dataland uploader can patch ..."

dann ist klar, dass du das für einen user testest, der die Dataland Uploader Rolle hat

"dataUploader" versteht man nur, wenn man weiß, dass das ein technical account ist

Reply:
done